multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+
+                    image_service_url = "http://lims2/cgi-bin/imageservice?path="
+                    zoom = 5        #need to check that this ACQ4 can support this size
+
+                    full_url = image_service_url + aff_image_path + "&zoom={}".format(str(zoom))
+                    print(full_url)
+                    save_path = self.base_path + '/' + jpg_image_name
+                    safe_save_path = lims.lims.safe_system_path(save_path)
+
+                    if os.path.exists(safe_save_path) == True:
+                        print("20x already moved")
+
+                    elif os.path.exists(safe_save_path) == False:
+                        image = urllib.URLopener()
+                        image.retrieve(full_url, safe_save_path)
+                        print("20x image moved")
+                    self.image_20 = safe_save_path
+                        
+                else:
+                    raise Exception("Couldn't find image in LIMS. Check you have the selected the correct slice.")
+                
+            except KeyError:
+                raise Exception('No Slice Selected')
+        else:
+            raise Exception('No Slice Selected')
+       
+
+
+    def create_lims_json(self):
+        """
+        Checks that markers are loaded into the canvas
+        Gets timestamp
+        Creates json for each cell clusters and saves locally
+        copies json to correct incoming folder
+        creates trigger file and drops in correct incoming folder
+        """
+        
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        
+        for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+
+        sites = set()
+        for item in items:
+            handle = item.opts.get('handle')
+            if handle is None:
+                continue
+            parent = handle.parent()
+            if parent.shortName().startswith('site_'):
+                sites.add(parent)
+
+        if len(sites) == 0:
+            raise Exception("No files loaded from site; can't determine which site to submit.")
+        if len(sites) > 1:
+            raise Exception("Files loaded from multiple sites; can't determine which site to submit.")
+        self.site_dh = list(sites)[0]
+        ts = self.site_dh.info()['__timestamp__']
+
+        
+        cluster_id = []
+        clusters = lims.cell_cluster_ids(self.slice_id)
+        for cid in clusters:
+            cluster_tmsp = lims.specimen_metadata(cid)['acq_timestamp']
+            if cluster_tmsp == ts:
+                cluster_id = cid
+                cluster_name = lims.specimen_name(cid)
+                print(cluster_name)
```

Comment:
It would be nice to raise an exception at this point if there were multiple matching clusters.

Reply:
ok